src/main/java/org/prebid/server/bidder/compass/CompassBidder.java

```diff
@@ -0,0 +1,173 @@
+        }
+    }
+
+    private ImpExtCompass getImpExtCompassWithType(ExtImpCompass extImpCompass) {
+        ImpExtCompass impExtCompass = null;
+        if (!Objects.equals(extImpCompass.getEndpointId(), "")) {
+            impExtCompass = ImpExtCompass.builder()
+                    .type("network")
+                    .endpointId(extImpCompass.getEndpointId())
+                    .build();
+        } else if (!Objects.equals(extImpCompass.getPlacementId(), "")) {
+            impExtCompass = ImpExtCompass.builder()
+                    .type("publisher")
+                    .placementId(extImpCompass.getPlacementId())
+                    .build();
+        }
+        return impExtCompass;
+    }
+
+    private ExtImpCompass parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), COMPASS_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final List<BidderError> errors = new ArrayList<>();
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse, errors), errors);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse,
+                                               List<BidderError> errors) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidRequest, bidResponse, errors);
+    }
+
+    private static List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse,
+                                                    List<BidderError> errors) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> resolveBidderBid(bid, bidResponse.getCur(), bidRequest.getImp(), errors))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private static BidderBid resolveBidderBid(Bid bid, String currency, List<Imp> imps, List<BidderError> errors) {
+        final BidType bidType;
+        try {
+            bidType = getBidType(bid.getImpid(), imps);
+        } catch (PreBidException e) {
+            errors.add(BidderError.badServerResponse(e.getMessage()));
+            return null;
```

Comment:
In the GO implementation, `makeBids` returns an error immediately, rather than collecting all the errors